docs/src/routes/(docs)/components/table-sortable/+page.svelte

```diff
@@ -1,42 +1,43 @@
-<script context="module">
+<script module>
   export const breadcrumb = "Sorteerbare tabel";
 </script>
 
-<script>
+<script lang="ts">
   import Code from "$lib/Code.svelte";
   import SideMenu from "$lib/SideMenu.svelte";
 
+  type Column = "firstname" | "lastname" | "age";
+
   const demoTableRows = [
     { firstname: "Janine", lastname: "Hinde", age: 16 },
     { firstname: "Henk", lastname: "de Vries", age: 8 },
     { firstname: "Maria", lastname: "Lin", age: 32 },
     { firstname: "John", lastname: "Doe", age: 64 },
   ];
 
-  /** @type {'firstname'|'lastname'|'age'|undefined} */
-  let sortColumn = undefined;
+  let sortColumn: Column | undefined = $state(undefined);
 
-  /** @type {'ascending'|'descending'} */
-  let sortDirection = "ascending";
+  let sortDirection: "ascending" | "descending" = $state("ascending");
```

Comment:
It maybe goes beyond this PR but when typing a value multiple times in a file (`ascending` and `descending`), I often move it to an enum, like:

`const enum SortDirection {
  Ascending = 'ascending',
  Descending = 'descending',
}`

or just variables like:
```
const ascending = 'ascending':
...
```

What do you think?

Reply:
Enums can have some benefits. They can be iterated on. They can help avoid extension of a type that's part of a strict API. And they can help with LSP refactoring.

But the nice thing about simple string literal union types is that they have 0 runtime overhead. So I tend to avoid unions unless I specifically need iteration or high strictness.